### PR TITLE
Fix build for re_outliner_entities

### PR DESCRIPTION
**Pull request by kaamaurice**
## Changelog Description
- Fix outliner_entity attr from container with get_root_outliner_datablocks()

### CODE REVIEW

File: openpype/hosts/blender/scripts/build_workfile.py [hunk 1]
```diff
@@ -386,7 +386,14 @@ def build_layout(project_name, asset_name):
 
     # Create layout instance
     layout_instance = create_instance("CreateLayout", "layoutMain")
-    layout_container = layout_instance.datablock_refs[0].datablock
+    layout_collection = next(
+        (
+            root
+            for root in layout_instance.get_root_outliner_datablocks()
```
Comment (Tilix4) [inline openpype/hosts/blender/scripts/build_workfile.py]: Better to use `get_datablocks(bpy.types.Collection)` without `if`

File: openpype/hosts/blender/scripts/build_workfile.py [hunk 2]
```diff
@@ -401,16 +408,13 @@ def build_layout(project_name, asset_name):
...
-                continue
-            if c.outliner_entity not in layout_container.children.values():
-                layout_container.children.link(c.outliner_entity)
-            if (
-                c.outliner_entity
-                in bpy.context.scene.collection.children.values()
-            ):
-                bpy.context.scene.collection.children.unlink(c.outliner_entity)
+        for container in containers:
+            for root in container.get_root_outliner_datablocks():
```
Comment (Tilix4) [inline openpype/hosts/blender/scripts/build_workfile.py]: Same here, if you're filtering only `Collection`, better to use `container.get_datablocks(Collection)`

File: openpype/hosts/blender/scripts/build_workfile.py [hunk 3]
```diff
@@ -401,16 +408,13 @@ def build_layout(project_name, asset_name):
...
+                if not isinstance(root, bpy.types.Collection):
```
Comment (Tilix4) [inline openpype/hosts/blender/scripts/build_workfile.py]: Here it looks like you've inverted the condition. Before it was continuing if not Collection, now it executes only if not Collection. I bet it is a mistake and you want to process only for Collection (cf my previous comment, you'll not need this any more)

File: openpype/hosts/blender/scripts/build_workfile.py [hunk 4]
```diff
@@ -560,8 +564,16 @@ def build_anim(project_name, asset_name):
         project_name, layout_repre, "LinkLayoutLoader"
     )
 
+    # keep layout root collection name before make container publishable.
+    layout_collection_name = next(
+        (
+            root.name
+            for root in layout_container.get_root_outliner_datablocks()
```
Comment (Tilix4) [inline openpype/hosts/blender/scripts/build_workfile.py]: Same

**Review by Tilix4 — CHANGES_REQUESTED** [review]
Nice, few suggestions regarding the new implementation

File: openpype/hosts/blender/scripts/build_workfile.py (hunk 3 above)
Comment (kaamaurice) [reply openpype/hosts/blender/scripts/build_workfile.py]: Ouch indeed !

**Review by Tilix4 — APPROVED** [review]
LGTM thanks!

**Pull request merged**